pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -80,22 +83,50 @@ func (pl *VolumeBinding) Name() string {
 	return Name
 }
 
-func (pl *VolumeBinding) podHasPVCs(pod *v1.Pod) bool {
+// podHasPVCs returns 2 values:
+// - the first one to denote if the given "pod" has any PVC defined.
+// - the second one to return any error if the requested PVC is illegal.
+func (pl *VolumeBinding) podHasPVCs(pod *v1.Pod) (bool, error) {
```

Comment:
this should return Status, error only if some internal state is bad, otherwise we should return unschedulable.

Reply:
ah, I see you use unschedulable when we return in PreFilter